envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/GroupChangeWatcher.kt

```diff
@@ -38,12 +39,13 @@ internal class GroupChangeWatcher(
         val groups = cache.groups()
         metrics.setCacheGroupsCount(groups.size)
         if (oldGroups != groups) {
-            emitNewGroupsEvent()
+            val symmetricDifference = Sets.symmetricDifference(oldGroups.toSet(), groups.toSet())
```

Comment:
After thinking it over, I think we shouldn't get symmetric difference here. We care only about new groups, that are added. Now we generate snapshots for just removed groups and we add them to cache again, which is counterproductive.